### PR TITLE
LE w/o Public IP

### PR DESCRIPTION
**Pull request by dzotic9**

### CODE REVIEW

File: manifest.jps [hunk 1]
```diff
@@ -41,18 +41,21 @@ targetNodes:
...
     - litespeedphp
+    - litespeedadc
+    - lemp
+    - llsmp
 
 requiredFeatures: extip
 
 homepage: https://github.com/jelastic-jps/lets-encrypt
-baseUrl: https://raw.githubusercontent.com/jelastic-jps/lets-encrypt/master
+baseUrl: https://raw.githubusercontent.com/dzotic9/lets-encrypt/master
```
Comment (SlavaKatiukha) [inline manifest.jps]:
```suggestion
baseUrl: https://raw.githubusercontent.com/jelastic-jps/lets-encrypt/master
```

File: manifest.jps [hunk 2]
```diff
@@ -81,6 +84,12 @@ settings:
     - type: string
       name: undeployHookType
       inputType: hidden
+    - type: list
+      name: withExtIp
+      default: true
+      values:
+        true: with ext IP
+        false: without ext IP
```
Comment (SlavaKatiukha) [inline manifest.jps]: remove UI

File: scripts/ssl-manager.js [hunk 3]
```diff
@@ -468,6 +472,89 @@ function SSLManager(config) {
...
+            resp;
+
+        if (config.bindedDomains) return config.bindedDomains;
+
+        resp = jelastic.env.binder.GetExtDomains(config.envName, session);
+        if (resp.result != 0) return resp;
+
+        for (var i = 0, n = resp.extDomains.length; i < n; i++) {
+            domain = resp.extDomains[i];
+            domains.push(domain.domain);
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js]: replace with
```domains.push(resp.extDomains[i].domain);```

File: scripts/ssl-manager.js [hunk 4]
```diff
@@ -468,6 +472,89 @@ function SSLManager(config) {
...
+                session: session,
+                extDomains: freeDomains.join(";")
+            });
+        }
+
+        return { result: 0 };
+    };
+
+    me.isBusyExtDomain = function (domain) {
+        var BUSY_RESULT = 2330,
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js]: hardcode.
Add ```import com.hivext.api.environment.response.EnvironmentResponse```
and replace with EnvironmentResponse.ENVIRONMENT_EXT_DOMAIN_IS_BUSY

File: scripts/ssl-manager.js [hunk 5]
```diff
@@ -831,6 +967,20 @@ function SSLManager(config) {
...
+    me.removeSSLCert = function removeSSLCert() {
+        var resp,
+            sslCerts;
+
+        resp = jelastic.env.binder.GetSSLCerts(config.envName, session);
+        if (resp.result != 0) return resp;
+
+        sslCerts = resp.responses;
+        // resp = jelastic.env.binder.RemoveExtDomain(config.envName, session, config.customDomains);
+        // if (resp.result != 0) return resp;
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js]: remove

File: scripts/ssl-manager.js [hunk 6]
```diff
@@ -1018,6 +1185,22 @@ function SSLManager(config) {
...
+        // };
+
+        // me.setBusyDomains = function(domains) {
+        //     config.busyDomains = domains;
+        // };
+        //
+        // me.getBusyDomains = function() {
+        //     return config.busyDomains || [];
+        // };
+
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js]: remove

File: scripts/ssl-manager.js [hunk 7]
```diff
@@ -468,6 +472,89 @@ function SSLManager(config) {
...
+            resp;
+
+        resp = jelastic.environment.binder.CheckExtDomain({
+            appid: config.envName,
+            session: session,
+            extdomain: domain
+        });
+
+        if (resp.result != 0 && resp.result != BUSY_RESULT) return resp;
+        return resp.result == BUSY_RESULT ? true : false;
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js]:
```suggestion
        return (resp.result == BUSY_RESULT);
```

File: manifest.jps (hunk 1 above)
Comment (dzotic9) [reply manifest.jps]: applied suggestion

File: scripts/create-installation-script.js [hunk 8]
```diff
@@ -10,6 +10,7 @@ var baseDir          = getParam("baseDir", "/"),
     deployHookType   = getParam("deployHookType", ""),
     undeployHook     = getParam("undeployHook", ""),
     undeployHookType = getParam("undeployHookType", ""),
+    withExtIp        = getParam("withExtIp", ""),
```
Comment (jelasticdev) [inline scripts/create-installation-script.js]:
```suggestion
    withExtIp        = getParam("withExtIp", "true"),
```

**Pull request merged**